modules/react/src/deckgl.js

```diff
@@ -84,6 +85,24 @@ export default class DeckGL extends React.Component {
     return this.pickObjects(opts);
   }
 
+  // Callbacks
+
+  // Forward callback and then call forceUpdate to guarantee that sub components update
+  _onResize(...args) {
+    if (this.props.onResize) {
+      this.props.onResize(...args);
+    }
+    this.forceUpdate();
+  }
+
+  // Forward callback and then call forceUpdate to guarantee that sub components update
+  _onViewStateChange(...args) {
+    if (this.props.onViewStateChange) {
+      this.props.onViewStateChange(...args);
```

Comment:
`viewState` contains `width` and `height` but `onViewStateChange` is not called when resize. This may cause discrepancy between the internal state and the app state.

Reply:
Hmm.... `width` and `height` in viewState doesn't really make sense any more as the entire system with view states and `Views` are designed to be size independent.

There is certainly a backwards compatibility aspect to think about, but maybe that can be mostly handled through the old `onViewportChange` system which doesn't support `initialViewState`?

I am not against adding an extra call somewhere. I'll open a separate issue to decide on it.